pkg/scheduler/scheduled_job_wrapper.go

```diff
@@ -0,0 +1,97 @@
+package scheduler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-co-op/gocron/v2"
+	"go.uber.org/zap"
+
+	"github.com/cms-enterprise/easi-app/pkg/logfields"
+	"github.com/cms-enterprise/easi-app/pkg/storage"
+)
+
+// ScheduledJobFunction is an abstraction around scheduled jobs and tasks it defines
+type ScheduledJobFunction[input comparable] func(context.Context, input)
+
+// RegisterJobFunction is a function that registers a job with the scheduler and returns the job
+type RegisterJobFunction func(context.Context, *storage.Store, gocron.Scheduler) (gocron.Job, error)
+
+type ScheduledJobWrapper[input comparable] struct {
+	name          string
+	jobDefinition gocron.JobDefinition
+	jobFunction   ScheduledJobFunction[input]
+	params        input
+	job           gocron.Job
+}
+type ScheduledJob struct {
+	ScheduledJobWrapper[*ScheduledJob] // Embed the wrapper with a pointer to ScheduledJob
+}
+
+func (sjw *ScheduledJobWrapper[input]) Register() {
+	RegisterJob(sjw.name, func(ctx context.Context, store *storage.Store, scheduler gocron.Scheduler) (gocron.Job, error) {
+		retJob, err := scheduler.NewJob(
+			sjw.jobDefinition,
+			gocron.NewTask(sjw.jobFunction, sjw.params),
+			// gocron.NewTask(sjw.jobFunction, sjw.params, scheduler),
```

Comment:
Still relevant?

Reply:
yes, this is a little bit of the refactor to see about passing dependencies not directly in context. I'll make sure to just leave one implementation in the refactor.